packages/flutter/lib/src/rendering/flow.dart

```diff
@@ -0,0 +1,345 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/animation.dart';
+import 'package:vector_math/vector_math_64.dart';
+
+import 'box.dart';
+import 'object.dart';
+
+/// A context in which a [FlowDelegate] paints.
+///
+/// Provides information about the current size of the container and the
+/// children and a mechanism for painting children.
+///
+/// See also:
+///
+///  * [FlowDelegate]
+///  * [Flow]
+///  * [RenderFlow]
+abstract class FlowPaintingContext {
+  /// The size of the container in which the children can be painted.
+  Size get size;
+
+  /// The number of children available to paint.
+  int get childCount;
+
+  /// The size of the [i]th child.
+  ///
+  /// If [i] is negative or exceeds [childCount], returns null.
+  Size getChildSize(int i);
+
+  /// Paint the [i]th child using the given transform.
+  ///
+  /// The child will be painted in a coordinate system that concatenates the
+  /// containers coordinate system with the given transform. The origin of the
+  /// parent's coordinate system is the upper left corner of the parent, with
+  /// x increasing rightward and y increasing downward.
+  ///
+  /// The container will clip the children to its bounds.
+  void paintChild(int i, Matrix4 transform);
+}
+
+/// A delegate that controls the appearance of a flow layout.
+///
+/// Flow layouts are optimized for moving children around the screen using
+/// transformation matrices. For optimal performance, construct the
+/// [FlowDelegate] with an [Animation] that ticks whenever the delegate wishes
+/// to change the transformation matrices for the children and avoid rebuilding
+/// the [Flow] widget itself every animation frame.
+///
+/// See also:
+///
+///  * [Flow]
+///  * [RenderFlow]
+abstract class FlowDelegate {
+  /// The flow will repaint whenever the [repaint] animation ticks.
+  const FlowDelegate({ Animation<Object> repaint }) : _repaint = repaint;
+
+  final Animation<Object> _repaint;
```

Comment:
i think i'd prefer dynamic to Object, since you don't care what the type is at all